### ISSUE TITLE
Filter deprecation

### ISSUE THREAD

**Issue by yann-eugone** [api-platform/core#3622]
**Description**  

Deprecations are [presented as replacement for api versioning](https://api-platform.com/docs/core/deprecations/).
ApiPlatform provide way to deprecate both Resource, Operation and Properties.

But I was not able to find any way to **deprecate Filters**, although this is often something that may evolve.

**Example** 

```diff
<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use ApiPlatform\Core\Annotation\ApiFilter;
use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\SearchFilter;
+ use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\RangeFilter;

/**
 * @ApiResource()
- * @ApiFilter(SearchFilter::class, properties={"id": "exact", "price": "exact", "description": "partial"})
+ * @ApiFilter(SearchFilter::class, properties={"description": "partial"})
+ * @ApiFilter(RangeFilter::class, properties={"price"})
+ * @ApiFilter(SearchFilter::class, properties={"id": "exact"}, deprecationReason="Use get operation instead")
+ * @ApiFilter(SearchFilter::class, properties={"price": "exact"}, deprecationReason="Use range filter instead")
 */
class Offer
{
}
```

The filter is still present, but should be deprecated using OpenAPI:v3 [Parameter Object](https://swagger.io/docs/specification/describing-parameters/) `deprecated` field name : 

![image](https://user-images.githubusercontent.com/1303838/85604904-0115c980-b652-11ea-818a-c5566756f376.png)

**Comment by soyuka** [comment]
will be fixed by #3407

**Comment by yann-eugone** [comment]
Good to known 😉
